### DISCUSSION TITLE
Graphical Audio Lounge Area

### DISCUSSION THREAD

**Discussion by gjaekel** [igniterealtime/openfire-pade-plugin#223] [Ideas]
I have the idea of a Graphical Audio Lounge Area to walk around and have small group talks. 

* Let the participants get together on a room as usual, lets say at 'Lounge'
* By help of some special room attribute, the client changes to the Lounge mode for all joining participants. This mode might be switched by the Moderator. But to allow to make it persistent, a well-known XMPP attribute should be used.
* Instead of a normal Jitsi-Client display, there's a "full-screen" 2D Canvas populated with the avatar circles of all participants. 
* Each participant may "move around" by click at some point of the canvas. Then, his avatar starts slow moving to this destination.
* If the own avatar is in a closer range to others, he's able to hear their audio. With raising distance, the audio level diminish dynamical down to about -20dB or more -- but maybe not to zero to have the imagination of a big room.
* Maybe the range should be visualized by a bigger outline circle of about something like 5 times the diameter of the Avatar circle size. It should be possible to "stand together" in small groups like in real life.
* To ease the grouping, the canvas may offer some static background image: A "floor plan" with some "bar tables", "couches" or something Lounge-like.
* This may be coupled with some trigger areas on the floor plan that represents additional breakout rooms. If a participant walk on that, a vanilla conference room change is initiated and the participant is passed to an vanilla Jitsi A/V conference to meet all other in this room. 

My basic technical idea for this is to use the local audio volume control feature as is it *already* *available* at each foreign participant window. To support the movement of participants, the already existing broadcast feature might be used to transport the destination coordinates.

I would appreciate it *very* much if one of this might be available -- in a first usable sprint -- for an upcoming online event, which is terminated for Mo, 2021-02-08.

**Comment by ClaudeStabile** [comment]
@gjaekel sounds very Interesting and would be a nice step. I was thinking about this since quite a while with a slighly different approach.
I was thinking about a kind of dungeons master with rooms corresponding on a map. Then you can move from one room to another if you gain access to it(Talking with people to get secret pass for next rooms etc..). Typically a dungeon master game with Padé/jitsi. Each user can be granted special power when it gain a level or a step. For exemple the possibility to generate extra rooms or invite people etc...

That would be a hudge job but could propose a very fun community Live with Vidconf, Chat meeting people and gaming. Of course it would require a serious sponsor to do such a project...
However you suggested an enhancement could be a nice very first step in this direction.

We can also imagine a service that could enable vidcall ou audio when you are in a certain location with GPS on your mobile. This could also bring some serious and fu app. Not on a virtal map but in real life too...

I fully support your suggestions/ideas

**Reply by gjaekel** [reply]
That's a whole game! There seem already to exist such an key-ready application in 2D-retro style (https://gather.town/); there's a demo access available.

I used it recently in real live as a participant of a small conference event (https://cyberland.ijug.eu/2021-01-ladies-night/). The organizers "build" a floor plan with a lobby and three "real" conference rooms with chairs and a stage. You may meet at the lounge (or everywhere) by staying close together. And you may switch to the Youtube stream of a session by taking a seat. BTW: The moderator and presenters and a small number of backstage audience (to suppoert the speakers no to speak into the off) were using Jitsi ans OBS.

**Comment by gjaekel** [comment]
Steps to do:
* Find a way to programmatically  access the local volume control available for each participant at the three-dots-menu of the participants video window. As a fallback, find a way to local(!) mute a certain participant.
* Add a "big" canvas (the lounge), switchable via icon by the moderator, appearing on each participants UI. In a later sprint: Load a custom background image to ease to choose places for coming together.
* Populate the canvas with the participants in an appropriate way (on first enable, with later joining/leaving participans): In later sprints: Maybe around the middle, not overlapping. Maybe with the avatar circles.
* Allow the participants to slowly move itself by aiming a destination with a click. In later sprints: Maybe avoid collitions.
* Calculate the participants local volume (or at least muting) by some appropriate function of distance to the one position.

* The participants video may be integrating by having a ribbon of small video windows at top or at the side the lounge map. The ribbon is filled with the video windows of the participants "in range". Maybe the the Filmstrip View can be used by toggling the visible state of the elements.

* As an extension, a feature may allow (for anyone in such a ranged group) to break out together in a new room (by help of the already existing technical breakout feature). For the remaining, there might be some "teleport" marker appear at this location to allow to enter this breakout rooms. Configurable optional, the avatars of the already teleported participants might be stay at the map as a "ghost" and/or the number of participants inside the breakout room may be shown.

* As an further extension, there might be a predefined set of such teleport markers. As an alternative, it might be allowed to break out for a single participant -- a moderator of a breakout session -- to spawn a teleport spot for others.

**Comment by gjaekel** [comment]
> Find a way to programmatically access the local volume ...

Might be `APP.UI.setAudioLevel(id, newLvl);`, see https://github.com/jitsi/jitsi-meet/blob/9895a0460911e39bbbc72b23fcbd0f6615d0edcb/conference.js#L1944-L1960
